coriolis/osmorphing/base.py

```diff
@@ -256,14 +259,19 @@ def _list_dir(self, chroot_path):
         path = os.path.join(self._os_root_dir, chroot_path)
         return utils.list_ssh_dir(self._ssh, path)
 
-    def _exec_cmd(self, cmd):
+    def _exec_cmd(self, cmd, timeout=0):
+        if timeout is not None:
+            timeout = timeout or self._osmorphing_operation_timeout
```

Comment:
This line will never allow timeout to be `0`, thus making the above `timeout is not None` check redundant.

We must convene that either:
* timeout of `0` is valid and settable, and thus should just make the default kwarg `None` and just write `if timeout is None: timeout = self._osmorphing_operation_timeout`
* timeout of `0` is semantically the same as `None` (i.e. no timeout), so we can just write it like `if not timeout: timeout = self._osmorphing_operation_timeout`

Reply:
I don't think we want to ever set paramiko channel timeout to 0. Usually, for no timeout, we should set the timeout to `None`. This way, the channels will be in 'blocking' mode - this is what we used before (more info here: https://github.com/paramiko/paramiko/blob/5cf145ffd3d7cd4d96a511376267a7f27cce5eed/paramiko/channel.py#L588).
I wanted to make a logic where `_exec_cmd` would have a way to disable the channel timeout (by setting it to `None`), and also ignore `self._osmorphing_operation_timeout`. If you don't think that's needed, I can just do what you said above.

Reply:
Anything works as long as we are internally consistent.

From my PoV having both `None` and `0` represent no timeout is excessive, though I'd recommend we make the kwarg itself here default to `None` to be clear that no timeout is set (a timeout of `0` could either mean no timeout ever or expecting each operation to be instantaneous so confusion might arise there)